backend/python/app/services/implementations/user_service_pg.py

```diff
@@ -0,0 +1,372 @@
+            reason = getattr(e, "message", None)
+            self.logger.error(
+                "Failed to create user. Reason = {reason}".format(
+                    reason=(reason if reason else str(e))
+                )
+            )
+            raise e
+        
+        new_user_dict = UserService.__user_to_dict_and_remove_auth_id(
+            new_user
+        )
+        new_user_dict["email"] = firebase_user.email
+        return UserDTO(**new_user_dict)
+
+    def update_user_by_id(self, user_id, user):
+        try:
+            old_user = User.query.get(user_id)
+
+            if not old_user:
+                raise Exception("user_id {user_id} not found".format(user_id=user_id))
+
+            test = User.query.filter_by(id=user_id).update(
+                {
+                    User.first_name: user.first_name,
+                    User.last_name: user.last_name,
+                    User.role: user.role
+                })
+
+            db.session.commit()
+
+            try:
+                firebase_admin.auth.update_user(old_user.auth_id, email=user.email)
+            except Exception as firebase_error:
+                try:
+                    old_user_dict = {
+                        User.first_name: old_user.first_name,
+                        User.last_name: old_user.last_name,
+                        User.role: old_user.role
+                    }
+                    User.query.filter_by(id=user_id).update(**old_user_dict)
+                    db.session.commit()
+
+                except Exception as postgres_error:
+                    reason = getattr(postgres_error, "message", None)
+                    error_message = [
+                        "Failed to rollback Postgres user update after Firebase user update failure.",
+                        "Reason = {reason},".format(
+                            reason=(reason if reason else str(postgres_error))
+                        ),
+                        "Postgres user id with possibly inconsitent data = {user_id}".format(
+                            user_id=user_id
+                        ),
+                    ]
+                    self.logger.error(" ".join(error_message))
+                
+                raise firebase_error
+        
+        except Exception as e:
+            reason = getattr(e, "message", None)
+            self.logger.error(
+                "failed to update user. Reason = {reason}".format(
+                    reason=(reason if reason else str(e))
+                )
+            )
+            raise e
+
+        return UserDTO(user_id, user.first_name, user.last_name, user.email, user.role)
+
+
+    def delete_user_by_id(self, user_id):
+        try:
+            deleted_user = User.query.get(user_id)
+
+            if not deleted_user:
+                raise Exception("user_id {user_id} not found".format(user_id=user_id))
+
+            delete_count = User.query.filter_by(id=user_id).delete()
```

Comment:
```suggestion
            delete_count = User.query.filter_by(id=user_id).delete(synchronize_session="fetch")
```

See https://docs.sqlalchemy.org/en/14/orm/session_basics.html#orm-expression-update-delete

Was seeing an error with "instance has been deleted or its row is otherwise not present" when deleting